### ISSUE TITLE
PerfManager alway return the same ProviderSummary

### ISSUE THREAD

**Issue by freebsdly** [vmware/govmomi#1204]
there are multi ManagedOjbets return by call finder.HostSystemList(), when i use Manager.ProviderSummary() to get mo's provider, it alway return the same.

 Because performance.Manager's providerSummary use map as cache, use entity's Type as key
i modifed to use Type + Value as key.
```
func (m *Manager) ProviderSummary(ctx context.Context, entity types.ManagedObjectReference) (*types.PerfProviderSummary, error) {
	m.providerSummary.Lock()
	defer m.providerSummary.Unlock()

	if m.providerSummary.m == nil {
		m.providerSummary.m = make(map[string]*types.PerfProviderSummary)
	}

	s, ok := m.providerSummary.m[entity.Type]
	if ok {
		return s, nil
	}

	req := types.QueryPerfProviderSummary{
		This:   m.Reference(),
		Entity: entity,
	}

	res, err := methods.QueryPerfProviderSummary(ctx, m.Client(), &req)
	if err != nil {
		return nil, err
	}

	s = &res.Returnval

	m.providerSummary.m[entity.Type+entity.Value] = s

	return s, nil
}
```

**Comment by dougm** [comment]
Don't recall why we had only used the object type as the key.  We could change the map to use MOR as key: `map[types.ManagedObjectReference]*types.PerfProviderSummary`

Or maybe we don't need a cache at all here?

**Comment by dougm** [comment]
Sounds good to me.  Do you want to open a PR that removes this cache?

**Comment by freebsdly** [comment]
i submit a PR already #1216

**Comment by dougm** [comment]
Fixed by #1216
